### PR TITLE
PLNSRVCE-251: fix matching of protected tags

### PR DESCRIPTION
**Pull request by Michkov**
Default repostory should be protected against pushing from other
namespaces. The image tag name should start with namespace.

https://github.com/redhat-appstudio/application-service/pull/129

### CODE REVIEW

**Comment by Michkov** [comment]
Cycling on:
```
1.6542451038371792e+09	ERROR	controllers.ComponentInitialBuild	Unable to create PipelineRun for {{ } {      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []} {<nil> <nil> [] []  []  <nil> nil <nil> [] []} {{0 [] map[]} {<nil> <nil> map[] map[] [] <nil> []}}}	{"Namespace": "test", "Application": "test-app", "Component": "component", "error": "invalid user image tag combination of default repo quay.io/redhat-appstudio/user-workload:mkovarik-component and component namespace test"}
github.com/redhat-appstudio/build-service/controllers.(*ComponentBuildReconciler).Reconcile
	/workspace/controllers/component_build_controller.go:135
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
1.654245103847257e+09	ERROR	controller.component	Reconciler error	{"reconciler group": "appstudio.redhat.com", "reconciler kind": "Component", "name": "component", "namespace": "test", "error": "invalid user image tag combination of default repo quay.io/redhat-appstudio/user-workload:mkovarik-component and component namespace test"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
1.6542451038632429e+09	ERROR	controllers.ComponentInitialBuild	Unable to create PipelineRun for {{ } {      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []} {<nil> <nil> [] []  []  <nil> nil <nil> [] []} {{0 [] map[]} {<nil> <nil> map[] map[] [] <nil> []}}}	{"Namespace": "test", "Application": "test-app", "Component": "component", "error": "invalid user image tag combination of default repo quay.io/redhat-appstudio/user-workload:mkovarik-component and component namespace test"}
```

**Comment by Michkov** [comment]
Tested by creating first component with protected component image -> pipelineRun was not created.
Then component recreated with non-protected component image -> pipelineRun created
```
1.6542495826820557e+09 INFO controller.pipelinerun Starting workers {"reconciler group": "tekton.dev", "reconciler kind": "PipelineRun", "worker count": 1}
1.6542495828233101e+09 INFO controllers.ComponentInitialBuild Waiting for devfile model in component: test/component {"ComponentInitialBuild": "test/component"}
1.6542495828283994e+09 INFO controllers.ComponentInitialBuild Waiting for devfile model in component: test/component {"ComponentInitialBuild": "test/component"}
1.6542495843470728e+09 INFO controllers.ComponentInitialBuild Waiting for devfile model in component: test/component {"ComponentInitialBuild": "test/component"}
1.6542495847523856e+09 ERROR controllers.ComponentInitialBuild Unable to create PipelineRun {"Namespace": "test", "Application": "test-app", "Component": "component", "error": "invalid user image tag combination of default repo quay.io/redhat-appstudio/user-workload:mkovarik-component and component namespace test"}
github.com/redhat-appstudio/build-service/controllers.(*ComponentBuildReconciler).Reconcile
/workspace/controllers/component_build_controller.go:135
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
1.6542496761310072e+09 INFO controllers.ComponentInitialBuild Waiting for devfile model in component: test/component {"ComponentInitialBuild": "test/component"}
1.6542496761360638e+09 INFO controllers.ComponentInitialBuild Waiting for devfile model in component: test/component {"ComponentInitialBuild": "test/component"}
1.6542496776555834e+09 INFO controllers.ComponentInitialBuild Waiting for devfile model in component: test/component {"ComponentInitialBuild": "test/component"}
1.6542496785467992e+09 INFO controllers.ComponentInitialBuild Initial build pipeline created for component component in test namespace {"Namespace": "test", "Application": "test-app", "Component": "component"}
```

File: controllers/component_build_controller.go [hunk 1]
```diff
@@ -133,16 +133,6 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 
 	if err := r.SubmitNewBuild(ctx, component); err != nil {
-		// Try to revert the annotation
```
Comment (mmorhun) [inline controllers/component_build_controller.go:136]: I would not give up on any error. SubmitNewBuild could fail due to different reasons and not all of them lead to infinite loop.

File: controllers/component_build_controller.go [hunk 2]
```diff
@@ -216,7 +206,11 @@ func (r *ComponentBuildReconciler) SubmitNewBuild(ctx context.Context, component
 	}
 
 	gitopsConfig := prepare.PrepareGitopsConfig(ctx, r.NonCachingClient, component)
-	initialBuild := gitops.GenerateInitialBuildPipelineRun(component, gitopsConfig)
+	initialBuild, err := gitops.GenerateInitialBuildPipelineRun(component, gitopsConfig)
+	if err != nil {
+		log.Error(err, "Unable to create PipelineRun")
+		return nil
```
Comment (mmorhun) [inline controllers/component_build_controller.go:212]: Why do we need to remove the annotation revert if `nil` is returned?

**Review by mmorhun — CHANGES_REQUESTED** [review]

**Review by mmorhun — APPROVED** [review]

**Comment by psturc** [comment]
Verified by creating a component with invalid (protected) container image URL. The PipelineRun wasn't created, and the error was logged only once. No further retries.

Then created a component with valid container image URL, verified that it created a PipelineRun that finished successfully.

**Pull request merged**